src/NetMPA.Catalog/NetMPA.Catalog.Api/Controllers/CategoryController.cs

```diff
@@ -72,23 +73,27 @@ public async Task<ActionResult> AddCategory([FromBody] CategoryAdd category)
         {
             if (category == null) return BadRequest();
 
-            await this.categoryService.Update(mapper.Map<Bll.Models.Category>(category));
+            await this.categoryService.Add(mapper.Map<Bll.Models.Category>(category));
+            
+            await this.bus.Publish<CategoryAdd>(category);
 
             return StatusCode(201);
         }
 
         /// <summary>
         /// Update Category
         /// </summary>
-        [HttpPut("{id}")]
+        [HttpPut]
```

Comment:
REST way is to use entity is from the URL, not from the body. If you need category to pass id to message bus you can create separate entity after updating

According to the task it is necessary to handle update product event. Nothing to do with category.